src/test/perl/path.t

```diff
@@ -108,9 +108,9 @@ sub verify_exception
     } else {
         ok(! $ec_check->error(), "Error reset after $msg");
     };
-    if ($noreset) {
+    if ($noreset && $mc->{fail}) {
```

Comment:
I didn't really modify existing tests but made a commit explictly dedicated to this as I think the test is wrong... I don't like that you may want to test/print something undefined. It doesn't make sense.